File: Framework/Utilites/Encrypt.cs
using Microsoft.JSInterop;
using System.Threading.Tasks;

namespace Encryption;
public class EncryptionService
{
    private readonly IJSRuntime _jsRuntime;

    public EncryptionService(IJSRuntime jsRuntime)
    {
        _jsRuntime = jsRuntime;
    }

    public async Task<string> EncryptString(string plainText, string secretKey)
    {
        Console.WriteLine("Encrypt in Encrypt.cs called");
        return await _jsRuntime.InvokeAsync<string>("encryptString", plainText, secretKey);
    }

    public async Task<string> DecryptString(string encryptedText, string secretKey)
    {
        return await _jsRuntime.InvokeAsync<string>("encrypt.decryptString", encryptedText, secretKey);
    }
}

File: Framework/wwwroot/encrypt.js
window.encryption = {
    encryptString: function (plainText, secretKey) {
        console.log('encryptString in encrypt.js called');
        return CryptoJS.AES.encrypt(plainText, secretKey).toString();
    },
    decryptString: function (encryptedText, secretKey) {
        var bytes = CryptoJS.AES.decrypt(encryptedText, secretKey);
        return bytes.toString(CryptoJS.enc.Utf8);
    }
};
